### PR TITLE
Hrysiuk add new currency in admin

### PR DESCRIPTION
**Pull request by TarasHrysiukk**
У AddNewCurrencyBL в методі registerNewCurrency 
не можу прописати  inputCodeCurrency(addNewCurrencyModel.getCurrencyCode());
 та 
inputValueCurrency(addNewCurrencyModel.getCurrencyValue());
просить зміти методи inputCodeCurrency та inputValueCurrency на int, якщо змінити то просить змінити на String.

### CODE REVIEW

File: src/main/java/com/opencart/datamodel/LoginAdminModel.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package com.opencart.datamodel;
+
+import lombok.Builder;
+import lombok.Getter;
+
+@Getter
+@Builder
+
```
Comment (adomanich) [inline src/main/java/com/opencart/datamodel/LoginAdminModel.java:8]: Please, remove extra space

File: src/main/java/com/opencart/datamodel/LoginAdminModel.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+public class LoginAdminModel {
+
+    private String adminUserName;
+    private String adminPassword;
+}
+
+
```
Comment (adomanich) [inline src/main/java/com/opencart/datamodel/LoginAdminModel.java:15]: Please, remove extra space

File: src/main/java/com/opencart/datamodel/NewCurrencyModel.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+        public AddNewCurrencyModel.Builder currencyCode(int value){
+            currencyCode = value;
+            return this;
+        }
+
+        public AddNewCurrencyModel.Builder currencyValue(int value){
+            currencyValue = value;
+            return this;
+        }
```
Comment (adomanich) [inline src/main/java/com/opencart/datamodel/NewCurrencyModel.java:35]: Please, remove commented code

File: src/main/java/com/opencart/driver/DriverRepository.java [hunk 4]
```diff
@@ -16,7 +16,7 @@
 
     public final static ThreadLocal<WebDriver> DRIVERS = new ThreadLocal<>();
     public static WebDriver webDriver;
-    private static String browserName = "Chrome";
+    private static String browserName = "Firefox";
```
Comment (adomanich) [inline src/main/java/com/opencart/driver/DriverRepository.java:19]: Revert it to default (Chrome), if you wanna use Firefox, please do it localy

File: src/main/java/com/opencart/pages/AdminPage.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+public class AdminPage extends BasePage{
+
+    @FindBy(xpath = ".//*[contains(@href , '#collapse7')]")
+    private WebElement systemButton;
+
+    @FindBy(xpath = ".//*[contains(@href , '#collapse7-2')]")
+    private WebElement localizationButton;
+
+    @FindBy(xpath = ".//*[contains(@href , 'http://localhost/opencart/upload/admin/index.php?route=localisation/currency&user_token=AG2XbPvlJfbHIeiYJAFVx0llKsGXueRi\n')]")
+    private WebElement currenciesButton;
```
Comment (adomanich) [inline src/main/java/com/opencart/pages/AdminPage.java:16]: Could you create a separate page and business logic for NavigationMenu functionality?

File: src/main/java/com/opencart/pages/AdminPage.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        return localizationButton;
+    }
+
+    public WebElement getCurrenciesButton() {
+        return currenciesButton;
+    }
+
+    public WebElement getAddNewCurrenciesButton() {
+        return addNewCurrenciesButton;
+    }
```
Comment (adomanich) [inline src/main/java/com/opencart/pages/AdminPage.java:35]: I think you need also create a separate CurrencyPage, for describing corresponding elements

File: src/main/java/com/opencart/repository/AddNewCurrencyRepository.java [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+import com.opencart.datamodel.NewCurrencyModel;
+import org.apache.commons.lang3.RandomStringUtils;
+
+public class AddNewCurrencyRepository {
+    private AddNewCurrencyRepository(){
+    }
+
+    public static NewCurrencyModel getAddNewCurrencyRepository(){
+       // int a = (int) Math.random();
```
Comment (adomanich) [inline src/main/java/com/opencart/repository/AddNewCurrencyRepository.java:11]: Please, remove commented code

File: src/main/java/com/opencart/repository/AdminLoginRepository.java [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+        return LoginAdminModel.builder()
+                .adminUserName(adminUser)
+                .adminPassword(adminPassword)
+                .build();
+
+
+    }
+
+
```
Comment (adomanich) [inline src/main/java/com/opencart/repository/AdminLoginRepository.java:29]: Please, remove extra spaces

File: src/main/java/com/opencart/steps/AdminPageBL.java [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    public AdminPageBL loginAdmin(){
+        LoginAdminModel adminPanelModel = AdminLoginRepository.getAdminPanelModel();
+        inputAdminUserName(adminPanelModel.getAdminUserName());
+        inputAdminPassword(adminPanelModel.getAdminPassword());
+        clickOnLoginAdminButton();
+
+        return this;
+    }
```
Comment (adomanich) [inline src/main/java/com/opencart/steps/AdminPageBL.java:22]: You need to create a separate AdminLoginPageBL business class for this logic

File: src/main/java/com/opencart/steps/MainPageBL.java [hunk 10]
```diff
@@ -9,4 +9,8 @@ public HeaderPageBL getHeaderPageBL() {
     public SearchProductPageBL getSearchPageBL() {
         return new SearchProductPageBL();
     }
+
+    public AdminPageBL getAdminPageBL() {
+        return new AdminPageBL();
+    }
```
Comment (adomanich) [inline src/main/java/com/opencart/steps/MainPageBL.java:15]: The admin page should not relate to MainPage in any ways

File: src/test/java/AddNewCurrencyTest.java [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+import static com.opencart.enums.URLs.ADMIN_URL;
+
+public class AddNewCurrencyTest  extends BaseTest{
+    @Test
+    public void addNewCurrency(){
+        new Navigation().navigateToUrl( ADMIN_URL.getValue());
+        MainPageBL mainPageBL = new MainPageBL();
+        AdminPageBL adminPageBL = mainPageBL.getAdminPageBL()
+                .loginAdmin();
+        Assert.assertEquals(1,1);
```
Comment (adomanich) [inline src/test/java/AddNewCurrencyTest.java:16]: Could you describe the value of this assertion - Assert.assertEquals(1,1)?

**Review by adomanich — CHANGES_REQUESTED** [review]

File: src/main/java/com/opencart/datamodel/NewCurrencyModel.java [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+
+import lombok.Builder;
+import lombok.Getter;
+
+@Getter
+@Builder
+public class NewCurrencyModel {
+    private String currencyTitle;
+    private int currencyCode;
+    private int currencyValue;
```
Comment (adomanich) [inline src/main/java/com/opencart/datamodel/NewCurrencyModel.java:11]: You can change is to string

File: src/main/java/com/opencart/steps/AddNewCurrencyBL.java [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+
+    public AddNewCurrencyBL() {
+        newCurrencyPage = new NewCurrencyPage();
+    }
+
+    public AddNewCurrencyBL registerNewCurrency(){
+        NewCurrencyModel addNewCurrencyModel = AddNewCurrencyRepository.getAddNewCurrencyRepository();
+        inputTitleCurrency(addNewCurrencyModel.getCurrencyTitle());
+        inputCodeCurrency(addNewCurrencyModel.getCurrencyCode());
+        inputValueCurrency(addNewCurrencyModel.getCurrencyValue());
```
Comment (adomanich) [inline src/main/java/com/opencart/steps/AddNewCurrencyBL.java:19]: Now it returns int, but inputCodeCurrency() should take a string

File: src/test/java/AddNewCurrencyTest.java (hunk 11 above)
Comment (TarasHrysiukk) [reply src/test/java/AddNewCurrencyTest.java:16]: this i wrote just to check if my test works

Comment (TarasHrysiukk) [reply src/test/java/AddNewCurrencyTest.java:16]: я написав це лише для того, щоб перевірити, чи працює мій тест

**Pull request closed without merging**